intellij/src/saros/intellij/ui/views/buttons/AddContactButton.java

```diff
@@ -0,0 +1,167 @@
+   * contact.
+   *
+   * <p>Calls {@link XMPPContactsService#addContact(JID, String, BiPredicate)} if the dialogs were
+   * not canceled.
+   */
+  private void showAddContactDialog() {
+
+    // TODO better communicate issue to user
+    // TODO use checkInput as it disables the OK button instead of just doing nothing when OK is
+    //  clicked on illegal input.
+    InputValidator inputValidator =
+        new InputValidator() {
+          @Override
+          public boolean checkInput(String inputString) {
+            return true;
+          }
+
+          @Override
+          public boolean canClose(String inputString) {
+            JID jid = new JID(inputString);
+
+            return jid.isValid() && !jid.getName().isEmpty();
+          }
+        };
+
+    String userID;
+
+    try {
+      userID =
+          SafeDialogUtils.showInputDialog(
+              project,
+              Messages.AddContactButton_contact_jid_dialog_message,
+              Messages.AddContactButton_contact_jid_dialog_initial_input,
+              Messages.AddContactButton_contact_jid_dialog_title,
+              inputValidator,
+              new TextRange(0, 0));
+
+    } catch (IllegalAWTContextException e) {
+      log.error("Account creation failed.", e);
+
+      // TODO show error?
+      // showAccountCreationFailedError(e);
+
+      return;
+    }
+
+    if (userID == null) {
+      log.debug("Adding contact canceled by user during user id entry.");
+
+      return;
+    }
+
+    JID jid = new JID(userID);
+
+    String nickname;
+
+    try {
+      nickname =
+          SafeDialogUtils.showInputDialog(
+              project,
+              Messages.AddContactButton_contact_nickname_dialog_message,
+              "",
+              Messages.AddContactButton_contact_nickname_dialog_title);
+
+    } catch (IllegalAWTContextException e) {
+      log.error("Account creation failed.", e);
+
+      // TODO show error?
+      // showAccountCreationFailedError(e);
+
+      return;
+    }
+
+    if (nickname == null) {
+      log.debug("Adding contact canceled by user during user nickname entry.");
+
+      return;
+    }
+
+    BiPredicate<String, String> dialog =
+        (title, message) -> {
+          log.fatal("showing dialog " + title + " - " + message);
+
+          Integer integer;
+
+          try {
+            integer = SafeDialogUtils.showYesNoDialog(project, message, title);
+          } catch (IllegalAWTContextException e) {
+            log.error("Tried to run dialog in an illegal context", e);
+
+            // TODO tell user?
+
+            return false;
+          }
+
+          return integer != null && integer.equals(com.intellij.openapi.ui.Messages.YES);
+        };
+
+    try {
+      xmppContactsService.addContact(jid, nickname, dialog);
+    } catch (OperationCanceledException e) {
+      // TODO
+      log.error("Adding contact failed", e);
```

Comment:
I do not think that this is an error here. The user has canceled on purpose or did I miss some details what is contained in the exception ?

Reply:
> The user has canceled on purpose or did I miss some details what is contained in the exception ?

Nope. I already adjusted this on my local version. I just wrote that before actually reading what the exception was coming from. Will just debug log and exit quietly.